pytorch_lightning/trainer/supporters.py

```diff
@@ -119,27 +122,251 @@ def mean(self):
         return self.total / self.num_values
 
 
+class LightningBatchSamplerWrapper:
+    """
+    This class wraps user batch sampler, so we can extract
+    the batch_indices for tracking each sample.
+    """
+
+    def __init__(self, batch_sampler):
+        self.batch_sampler = batch_sampler
+        self.batch_indices = None
+
+    def __iter__(self):
+        for batch_indices in self.batch_sampler:
+            self.batch_indices = batch_indices
+            yield batch_indices
+
+    @staticmethod
+    def to_new_dataloader(dataloader: DataLoader) -> DataLoader:
```

Comment:
This function is duplicated in `trainer/data_loading.py`

can you reuse it? or reuse this one

Reply:
I tried, but it was starting to be more ugly than helpful. The code is very similar, but it does different things.

Reply:
so maybe split it into smaller functions and re-user on the same functionalities?